collection/lens.rkt

```diff
@@ -39,6 +67,21 @@
   (make-lens (λ (target) (nth target index))
              (λ (target value) (set-nth target index value))))
 
+(define (mapper-lens lens)
```

Comment:
Looking at this makes me think `map-lens` makes sense. I was initially thinking of the `map` in `map-lens` as a verb, which would make it something like `(lens-map first-lens '((1 2) (3 4) (5 6)))`, but as a noun `map-lens` works just fine.